engine/apps/slack/client.py

```diff
@@ -3,38 +3,37 @@
 
 from django.utils import timezone
 from slackclient import SlackClient
-from slackclient.exceptions import TokenRefreshError
 
 from apps.slack.constants import SLACK_RATE_LIMIT_DELAY
 
-from .exceptions import (
-    SlackAPIChannelArchivedException,
-    SlackAPIException,
-    SlackAPIRateLimitException,
-    SlackAPITokenException,
-)
-from .slack_client_server import SlackClientServer
-
 logger = logging.getLogger(__name__)
 
 
-class SlackClientWithErrorHandling(SlackClient):
-    def __init__(self, token=None, **kwargs):
-        """
-        This method is rewritten because we want to use custom server SlackClientServer for SlackClient
-        """
-        super().__init__(token=token, **kwargs)
+class SlackAPIException(Exception):
+    def __init__(self, *args, **kwargs):
+        self.response = {}
+        if "response" in kwargs:
+            self.response = kwargs["response"]
+        super().__init__(*args)
 
-        proxies = kwargs.get("proxies")
 
-        if self.refresh_token:
-            if callable(self.token_update_callback):
-                token = None
-            else:
-                raise TokenRefreshError("Token refresh callback function is required when using refresh token.")
-        # Slack app configs
-        self.server = SlackClientServer(token=token, connect=False, proxies=proxies)
```

Comment:
`self.server` is the _only_ spot where `SlackClientServer` is referenced, and `SlackClientWithErrorHandling.server` is never referenced, hence we can remove the class.

Additionally, `SlackClientWithErrorHandling.__init__` was only ever overriden to override `self.server`, with that said we can remove the overriden `__init__`.

Reply:
btw we're using `slackclient==1.3.0`, and the latest version is `3.21.3`. Looks like the scenario when `json.loads` fails [is handled better](https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/web/base_client.py#L298-L302) in the latest version. should we consider upgrading?

Reply:
[1.3.0](https://github.com/slackapi/python-slack-sdk/releases/tag/v1.3.0) is 5 years old 😮

Reply:
good catch 👍 I'll take a look at migrating us to the latest version